cpp/src/arrow/engine/substrait/function_test.cc

```diff
@@ -574,6 +575,8 @@ struct AggregateTestCase {
   std::shared_ptr<DataType> output_type;
   // The aggregation takes zero columns as input
   bool nullary = false;
+  // The aggregation is ordered
+  bool ordered = false;
```

Comment:
If aggregation is ordered, shouldn't this be true?

Reply:
The pre-PR behavior is equivalent to post-PR behavior with `ordered = false`. In pre-PR, all tested aggregators had no trouble with the default executor. In post-PR, the `first` and `last` aggregators are ordered and with `ordered = false` there is an error `NotImplemented: Using ordered aggregator in multiple threaded execution is not supported`.

Reply:
Can you add some comment to make this more clear? Just looking at
```
  // The aggregation is ordered
  bool ordered = false;
```
It's kind of confusing

Reply:
Clarified.